glean-core/android/src/main/java/mozilla/telemetry/glean/private/StringListMetricType.kt

```diff
@@ -98,11 +135,24 @@ class StringListMetricType(
      */
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testGetValue(pingName: String = sendInPings.first()): List<String> {
-        // @Suppress("EXPERIMENTAL_API_USAGE")
-        // Dispatchers.API.assertInTestingMode()
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        if (!testHasValue(pingName)) {
+            throw NullPointerException()
+        }
 
-        // return StringListsStorageEngine.getSnapshot(pingName, false)!![identifier]!!
-        assert(false, { "Testing API not implementated for StringListMetricType" })
-        return listOf("FOO", "BAR")
+        val jsonRes: JSONArray
+        val ptr = LibGleanFFI.INSTANCE.glean_string_list_test_get_value_as_string(Glean.handle, this.handle, pingName)!!
+        try {
+            jsonRes = JSONArray(ptr.getAndConsumeRustString())
+        } catch (e: org.json.JSONException) {
+            throw NullPointerException()
+        }
+        val stringList: MutableList<String> = mutableListOf()
+        for (i in 0..(jsonRes.length() - 1)) {
+            stringList.add(jsonRes.getString(i))
+        }
```

Comment:
Kotlin has this neat/horrifying feature that lets you add new methods to external classes.  In android-components, they use that to add [utility methods](https://github.com/mozilla-mobile/android-components/blob/742786c25f88ce8ab9dd3203a04104082e67eca0/components/support/ktx/src/main/java/mozilla/components/support/ktx/android/org/json/JSONArray.kt#L34) on top of the `org.json` library.  We could do something like that here.

Even if we don't go that far, maybe we could use the functional approach [in this toList function](https://github.com/mozilla-mobile/android-components/blob/742786c25f88ce8ab9dd3203a04104082e67eca0/components/support/ktx/src/main/java/mozilla/components/support/ktx/android/org/json/JSONArray.kt#L34).

Reply:
Oh geez, that's... well that's a thing isn't it. I'm not a Kotlin dev by trade, so I'll bow to whatever is most Kotlin-idiomatic.

That being said, where does `asSequence` come from? I skim-read the [`JSONArray` docs](https://developer.android.com/reference/kotlin/org/json/JSONArray) and couldn't find any way to iterate over it except taking the length and doing a ranged `for`.

Reply:
Oh, geeze, it looks like `asSequence` is a [a-c custom thing](https://github.com/mozilla-mobile/android-components/blob/742786c25f88ce8ab9dd3203a04104082e67eca0/components/support/ktx/src/main/java/mozilla/components/support/ktx/android/org/json/JSONArray.kt#L13), too.  Doing some sort of iterator magic.  Given all that, I'd say what you've got here is more than good enough...

Reply:
We're [importing a-c support libraries](https://github.com/mozilla/glean/blob/ff85c3cfc969668123de95a83ebd468f3821c68b/glean-core/android/build.gradle#L135), so we could use it ;)

Reply:
If it really is easy to use, we should totally do that (do we currently depend on any other imported functionality from that lib?)

@Dexterp37: Could you advise chutten on what to import here?

If that doesn't work out though staying with the loop here doesn't seem too bad (afterall it's the test function)

Reply:
I think this could be as easy as `jsonRes.toList<String>()` (see [the test code](https://github.com/mozilla-mobile/android-components/blob/742786c25f88ce8ab9dd3203a04104082e67eca0/components/support/ktx/src/test/java/mozilla/components/support/ktx/android/org/json/JSONArrayTest.kt#L59))

Reply:
Ah, but this might require an `import mozilla.components.support.ktx.android.org.json.toList` in this package right?

Reply:
Yup, shouldn't be too much of a big deal. We're already importing the dependency

Reply:
Cool.  I forgot we had that dependency already.  In that case, let's use it.